dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/Main.java

```diff
@@ -53,14 +57,36 @@
     private ResourceBundle resourceBundle = null;
     private Stage primaryStage = null;
     private JobInterruptionStore jobInterruptionStore = null;
-    private final EventHandler<WindowEvent> confirmCloseEventHandler = event -> {
 
-        if (jobWorkers.getTasks().isEmpty()) {
-            closeApplication();
-            event.consume();
 
-        } else {
+    /*************************/
+    private static final String WINDOW_POSITION_X = "Window_Position_X";
+    private static final String WINDOW_POSITION_Y = "Window_Position_Y";
+    private static final String WINDOW_WIDTH = "Window_Width";
+    private static final String WINDOW_HEIGHT = "Window_Height";
+    private static final double DEFAULT_X = 10;
+    private static final double DEFAULT_Y = 10;
+    private static final double DEFAULT_WIDTH = 800;
+    private static final double DEFAULT_HEIGHT = 600;
+    private static final String NODE_NAME = "ViewSwitcher";
+    private static final String BUNDLE = "Bundle";
+
+    /*************************/
+
+    Preferences pref = Preferences.userRoot().node(NODE_NAME);
+    double x = pref.getDouble(WINDOW_POSITION_X, DEFAULT_X);
+    double y = pref.getDouble(WINDOW_POSITION_Y, DEFAULT_Y);
+    double width = pref.getDouble(WINDOW_WIDTH, DEFAULT_WIDTH);
+    double height = pref.getDouble(WINDOW_HEIGHT, DEFAULT_HEIGHT);
+
+    private final EventHandler<WindowEvent> confirmCloseEventHandler = event -> {
 
+        //running tasks which are not in cache
+        List<Ds3JobTask> notCachedRunningTasks = jobWorkers.getTasks().stream().filter(task -> task.getProgress() != 1).collect(Collectors.toList());
```

Comment:
This looks like it can be made final.  To conform with our coding standard, all variables that are not reassigned must be made final.  Here is a link to our [coding standards](https://github.com/SpectraLogic/spectralogic.github.com/wiki/Java-Style-Guide#use-of-final-keyword).